src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs

```diff
@@ -271,12 +271,20 @@ private void WriteStringValue(DateTimeOffset value, ref int idx)
 
         private void FormatLoop(DateTimeOffset value, ref int idx)
         {
-            if (!Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out int bytesWritten, s_dateTimeStandardFormat))
+            Span<byte> tempSpan = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            bool result = Utf8Formatter.TryFormat(value, tempSpan, out int bytesWritten, s_dateTimeStandardFormat);
+            Debug.Assert(result);
+
+            JsonWriterHelper.TrimDateTimeOffset(tempSpan.Slice(0, bytesWritten), out bytesWritten);
+
+            if (_buffer.Slice(idx).Length < bytesWritten)
```

Comment:
nit: just realized this slice is unnecessary and duplicated in the common case.

Maybe:
```C#
if (_buffer.Length - idx < bytesWritten)
```